File: Leetcode/count_of_matches_in_tournament/count_of_matches.py
class Solution(object):
    def numberOfMatches(self, n):
        matches = 0
        while n > 1:
            if n % 2 == 0:
                matches += n // 2
                n //= 2
            else:
                matches += (n - 1) // 2
                n = (n - 1) // 2 + 1
        return matches

# Test cases
sol = Solution()
print(sol.numberOfMatches(7))  # Output: 6
print(sol.numberOfMatches(14))  # Output: 13

File: Leetcode/count_of_matches_in_tournament/count_of_matches.rb
def number_of_matches(n)
  matches = 0
  while n > 1
    if n.even?
      matches += n / 2
      n = n / 2
    else
      matches += (n - 1) / 2
      n = (n - 1) / 2 + 1
    end
  end
  matches
end

p number_of_matches(7)
p number_of_matches(14)
